### PR TITLE
Fix handling of arm64 instruction sets in ilc

### LINKED ISSUE #68132: ILCompiler does not recognize "sha1" and "sha2" Instruction Sets
Repro:

```cs
using System;

namespace aotRepro
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine($"base: {System.Runtime.Intrinsics.Arm.ArmBase.IsSupported}");
            Console.WriteLine($"neon: {System.Runtime.Intrinsics.Arm.AdvSimd.IsSupported}");
            Console.WriteLine($"aes: {System.Runtime.Intrinsics.Arm.Aes.IsSupported}");
            Console.WriteLine($"crc: {System.Runtime.Intrinsics.Arm.Crc32.IsSupported}");
            Console.WriteLine($"dotprod: {System.Runtime.Intrinsics.Arm.Dp.IsSupported}");
            Console.WriteLine($"rdma: {System.Runtime.Intrinsics.Arm.Rdm.IsSupported}");
            Console.WriteLine($"sha1: {System.Runtime.Intrinsics.Arm.Sha1.IsSupported}");
            Console.WriteLine($"sha2: {System.Runtime.Intrinsics.Arm.Sha256.IsSupported}");			
        }
    }
}
```

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <RuntimeIdentifier>win-arm64</RuntimeIdentifier>
     <PlatformTarget>ARM64</PlatformTarget>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.DotNet.ILCompiler" Version="7.0.0-*" />
    <IlcArg Include="--instructionset:sha1,sha2" />
  </ItemGroup>
</Project>
```

```cmd
dotnet run -c Release
```

```log
base: True
neon: True
aes: True
crc: True
dotprod: False
rdma: False
sha1: True
sha2: True
```

```cmd
dotnet publish -c Release
```

```log
  Generating compatible native code. To optimize for size or speed, visit https://aka.ms/OptimizeCoreRT
EXEC : error : Specified method is not supported. [C:\Projects\repro\aotRepro\aotRepro.csproj]
  System.NotSupportedException: Specified method is not supported.
     at ILCompiler.HardwareIntrinsicHelpers.Arm64IntrinsicConstants.FromInstructionSetFlags(InstructionSetFlags instruc
  tionSets)
     at ILCompiler.ExpectedIsaFeaturesRootProvider.ILCompiler.ICompilationRootProvider.AddCompilationRoots(IRootingServ
  iceProvider rootProvider)
     at ILCompiler.Compilation..ctor(DependencyAnalyzerBase`1 dependencyGraph, NodeFactory nodeFactory, IEnumerable`1 c
  ompilationRoots, ILProvider ilProvider, DebugInformationProvider debugInformationProvider, DevirtualizationManager de
  virtualizationManager, IInliningPolicy inliningPolicy, Logger logger)
     at ILCompiler.ILScannerBuilder.ToILScanner()
     at ILCompiler.Program.Run(String[] args)
     at ILCompiler.Program.Main(String[] args)
```

FWIW I took the names from https://github.com/dotnet/runtime/blob/ce61c09a5f6fc71d8f717d3fc4562f42171869a0/src/coreclr/tools/Common/JitInterface/CorInfoInstructionSet.cs#L739-L740

cc @jkotas

### PR DESCRIPTION
**Pull request by jkotas**
- Make rcpc instruction Specifiable and regenerated all auto-generated files
- Add missing cases for more recently added instruction sets in various places
- Delete some redundant logic

Fixes #68132

### CODE REVIEW

File: src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/HardwareIntrinsicHelpers.Aot.cs [hunk 1]
```diff
@@ -97,149 +101,96 @@ private static class XArchIntrinsicConstants
...
-                    {
-                        InstructionSet.ARM64_ArmBase => ArmBase,
-                        InstructionSet.ARM64_ArmBase_Arm64 => ArmBase_Arm64,
-                        InstructionSet.ARM64_AdvSimd => AdvSimd,
-                        InstructionSet.ARM64_AdvSimd_Arm64 => AdvSimd_Arm64,
-                        InstructionSet.ARM64_Aes => Aes,
-                        InstructionSet.ARM64_Crc32 => Crc32,
-                        InstructionSet.ARM64_Crc32_Arm64 => Crc32_Arm64,
-                        InstructionSet.ARM64_Sha1 => Sha1,
-                        InstructionSet.ARM64_Sha256 => Sha256,
```
Comment (jkotas) [inline src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/HardwareIntrinsicHelpers.Aot.cs:232]: Missing cases for `ARM64_Sha1_Arm64` and `ARM64_Sha256_Arm64` here were the direct cause of the crash from #68132

File: src/coreclr/nativeaot/Runtime/windows/PalRedhawkMinWin.cpp [hunk 2]
```diff
@@ -554,12 +554,6 @@ REDHAWK_PALIMPORT void REDHAWK_PALAPI PAL_GetCpuCapabilityFlags(int* flags)
 {
     *flags = 0;
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/windows/PalRedhawkMinWin.cpp:556]:
```suggestion

    // FP and SIMD support are enabled by default
    *flags |= ARM64IntrinsicConstants_AdvSimd;
    *flags |= ARM64IntrinsicConstants_AdvSimd_Arm64;
```

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you @jkotas !

It would be great to include this fix before we snap for Preview 4, as it would allow me to perform a better .NET 7 vs .NET 7 NativeAOT perf comparison on arm64 while working on Preview 4 perf report.

**Comment by adamsitnik** [comment]
@jkotas I wanted to tests these changes and realized that new NuGet package has not been published yet. This is perfectly fine, but I just wanted to ask how frequently are we releasing a new version of `Microsoft.Dotnet.ILCompiler` package? Is this a manual process?

**Comment by jkotas** [comment]
`Microsoft.Dotnet.ILCompiler` package is part of the dotnet/runtime official build, same as everything else in dotnet/runtime repo. The official builds should be running twice a day.

If you do not see `Microsoft.Dotnet.ILCompiler` package published, it means that we did not have a successful official build with these changes yet.

**Comment by jkotas** [comment]
I see `7.0.0-preview.4.22219.1` on the .NET 7 nightly feed (https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet7/nuget/v3/index.json) that has these changes.

**Pull request merged**